db/schema_applier.cc

```diff
@@ -628,87 +628,64 @@ static future<affected_tables_and_views> merge_tables_and_views(distributed<serv
-        });
+        diff.new_token_metadata = co_await db.local().get_notifier().prepare_tablet_metadata(tablet_hint);
     }
 
-    co_await db.invoke_on_all([&] (replica::database& db) -> future<> {
-        // In order to avoid possible races we first create the tables and only then the views.
-        // That way if a view seeks information about its base table it's guaranteed to find it.
-        co_await max_concurrent_for_each(diff.tables.created, max_concurrent, [&] (global_schema_ptr& gs) -> future<> {
-            co_await db.add_column_family_and_make_directory(gs, replica::database::is_new_cf::yes);
-        });
-        co_await max_concurrent_for_each(diff.views.created, max_concurrent, [&] (global_schema_ptr& gs) -> future<> {
-            co_await db.add_column_family_and_make_directory(gs, replica::database::is_new_cf::yes);
-        });
-    });
-    co_await db.invoke_on_all([&](replica::database& db) -> future<> {
-        diff.columns_changed.reserve(diff.tables.altered.size() + diff.views.altered.size());
-        for (auto&& altered : boost::range::join(diff.tables.altered, diff.views.altered)) {
-            diff.columns_changed.push_back(db.update_column_family(altered.new_schema));
-            co_await coroutine::maybe_yield();
-        }
-    });
-
-    // Insert column_mapping into history table for altered and created tables.
-    //
-    // Entries for new tables are inserted without TTL, which means that the most
-    // recent schema version should always be available.
-    //
-    // For altered tables we both insert a new column mapping without TTL and
-    // overwrite the previous version entries with TTL to expire them eventually.
-    //
-    // Drop column mapping entries for dropped tables since these will not be TTLed automatically
-    // and will stay there forever if we don't clean them up manually
-    co_await max_concurrent_for_each(diff.tables.created, max_concurrent, [&proxy] (global_schema_ptr& gs) -> future<> {
-        co_await store_column_mapping(proxy, gs.get(), false);
-    });
-    co_await max_concurrent_for_each(diff.tables.altered, max_concurrent, [&proxy] (schema_diff::altered_schema& altered) -> future<> {
-        co_await when_all_succeed(
-            store_column_mapping(proxy, altered.old_schema.get(), true),
-            store_column_mapping(proxy, altered.new_schema.get(), false));
-    });
-    co_await max_concurrent_for_each(diff.tables.dropped, max_concurrent, [&sys_ks] (schema_diff::dropped_schema& dropped) -> future<> {
-        schema_ptr s = dropped.schema.get();
-        co_await drop_column_mapping(sys_ks.local(), s->id(), s->version());
-    });
-
     co_return diff;
 }
 
+global_schema_diff schema_diff::global() const {
+    global_schema_diff g;
+    for (const auto& c : created) {
+        g.created.emplace_back(c);
+    }
+    for (const auto& a : altered) {
+        g.altered.push_back(global_schema_diff::altered_schema{
```

Comment:
nit: space before '{'

Reply:
I'll fix it but it's pre-existing thing, in general we can't expect one would fix every pre-existing thing, during code review we usually skip such things if they already exist in the code otherwise review gets convoluted.

Reply:
btw since it's constructor I guess it should be the other way around, few lines below there should not be a space added

Reply:
yep, my bad, it's a constructor call.